apps/i2c-accelerometer-gyroscope/src/i2c-accelerometer-gyroscope.c

```diff
@@ -0,0 +1,150 @@
+ * velocity from a gy-251 sensor.
+ *
+ */
+
+#include <zephyr.h>
+#include <net/net_core.h>
+#include <logging/log.h>
+#include <device.h>
+#include <i2c.h>
+
+#include "knot.h"
+#include <knot/knot_types.h>
+#include <knot/knot_protocol.h>
+
+/* I2C slave address */
+#define MPU_ADDRESS			0X68
+
+/* MPU6050 internal registers */
+#define MPU6050_REG_PWR_MGMT1		0x6B
+#define MPU6050_SLEEP_EN		BIT(6)
+#define MPU6050_REG_GYRO_CFG		0x1B
+#define MPU6050_REG_ACCEL_CFG		0x1C
+#define MPU6050_REG_DATA_START		0x3B
+
+/* Full Scale Range to +- 2g */
+#define FS_SEL				0x00
+/* Full Scal Range to 250deg/seg */
+#define AFS_SEL				0x00
+
+/* Time interval (in mseg) to get new measures */
+#define TIME_INTERVAL			200
+
+/*
+ * Note that measured data has 16 bits, what makes necessary to put together
+ * the MSB's with their LSB's.
+ * Example:
+ * buf[0] will have MSB's for x axis acceleration and buf[1] will have LSB's
+ * for x axis acceleration, that is, the raw data of x axis acceleration will
+ * be buf16[0] = (buf[0] << 8) | buf(1)
+ */
+#define MERGE_MSB_LSB(a,b) ( a << 8 | b )
+
+/*
+ * Offset values for sensor calibration
+ * To obtain these values, put the sensor in your chosen origin and realise a
+ * group of measurements, then take the mean value for each variable.
+ * Obs: Note that if you're using a high precision sensibility, the measures
+ * will vary slightly everytime.
+ */
+
+#define ACCEL_X_OFFSET		236
+#define ACCEL_Y_OFFSET		-752
+#define ACCEL_Z_OFFSET		20996
+#define GYRO_X_OFFSET		75
+#define GYRO_Y_OFFSET		-147
+#define GYRO_Z_OFFSET		23
+
+/* Buffer vector to store data from sensor 8 bit registers */
+u8_t register_buffer[14];
+s16_t measured_data[7]; /* Buffer vector to store measured data from sensor */
+
+s32_t accel[3], gyro[3]; /* Each vector position corresponds to an axis, e.g.
+accel[0] -> acceleration in x axis */
+
+LOG_MODULE_REGISTER(accelgyro, LOG_LEVEL_DBG);
+
+struct device *i2c_dev;
+
+void setup(void)
+{
+	/* Peripherals control */
+	i2c_dev = device_get_binding(DT_NORDIC_NRF_I2C_0_LABEL);
+	/* Checks if peripheral was properly configured */
+	if (!i2c_dev) {
+		printk("I2C: Device driver not found.\n");
+
+		return;
+	}
+	/* Take sensor out of sleep mode */
+	i2c_reg_update_byte(i2c_dev, MPU_ADDRESS,
+				MPU6050_REG_PWR_MGMT1, MPU6050_SLEEP_EN, 0);
+	/* Set the full scale range for gyroscope */
+	i2c_reg_write_byte(i2c_dev, MPU_ADDRESS, MPU6050_REG_GYRO_CFG, FS_SEL);
+	/* Set the full scale range for accelerometer */
+	i2c_reg_write_byte(i2c_dev, MPU_ADDRESS,
+				MPU6050_REG_ACCEL_CFG, AFS_SEL);
+
+	char *name_list[6] = { "ACCELX", "ACCELY", "ACCELZ",
+				"GYROX", "GYROY", "GYROZ" };
+	/* KNoT config */
+	for (int i = 0; i<3; i++) {
+		knot_data_register(i, name_list[i], KNOT_TYPE_ID_ACCELERATION
+					KNOT_VALUE_TYPE_INT,
+					KNOT_UNIT_ACCELERATION_MS2,
+					&accel[i], sizeof(accel[i]),
+					NULL, NULL);
+		knot_data_config(i, KNOT_EVT_FLAG_CHANGE, NULL);
+	}
+	for (int k = 0; k<3; k++) {
```

Comment:
You do not need to declare the variable `k` since you can use the variable `i` as the iterator of this loop. The review comments in the previous loop statement are valid for this one also.

Reply:
Done